### ISSUE TITLE
Functionality to add Actions for Subtab Data

### ISSUE THREAD

**Issue by metas-mk** [metasfresh/metasfresh-webui-frontend#982]
### Is this a bug or feature request?
Feature Request
### What is the current behavior?
Currently, it is not able to add Actions to Subtab data/ rows.
#### Which are the steps to reproduce?
Open the window/540112/. In Action menu there is one action available. In Swing Client there are further Actions available for the included Subtab.
### What is the expected or desired behavior?
We need a possibility to also start actions for the Subtab. Ideas are
a) Context Menu (right click) on subtab row and show the action there. (downside: can become very overcrowded, the user must use the mouse)
b) allow the subtab actions to be shown in main view action menu, only when subtab row is selected (downside: tough to understand for the user when an action is shown and when not)
c) Show an action button/ quickaction dropdown in modal overlay (advanced edit of subtab data) for the user. It could be placed somewhere left to the Done Button.

Currently, I prefer solution c) thx @teosarca ;-) 
@cadavre WDYT?

Required for https://github.com/metasfresh/metasfresh/issues/2074

**Comment by cadavre** [comment]
If actions are not frequently used I'm also for c). But on the other side it doesn't need to exclude a) implemented. Right now we have there "Ref docs" – we can also add actions for users who use mouse and wants to take actions faster (than opening Advanced edit, waiting for modal to load and then pressing on action).

#### I would also propose to add subcontext to context menus:
![subcontext](https://user-images.githubusercontent.com/461796/27864888-b847e99a-6190-11e7-8129-918a7ef0ec20.jpg)

This way we can:
1. Save API calls because if user wouldn't need to open i.e. ref docs – on right mouse click he will see regular menu w/out making API calls in first place.
2. Add more elements to context menu because they will be grouped.

Basically – this subcontext menus could even be placed in another issue as Feature request.

**Comment by metas-mk** [comment]
Additional Info: Asked selected Users about the 3 Options, b) would be their preferred solution.

b) allow the subtab actions to be shown in main view action menu, only when subtab row is selected

**Comment by teosarca** [comment]
in case of b), i think we need to make it obvious when user is opening the Action menu.
e.g. show a record description or something, above the Actions.

**Comment by metas-mk** [comment]
Where shall the new subtab action be placed (see Sub Action 1+2 depending of subtab selection)?
![image](https://user-images.githubusercontent.com/13365687/28066957-65c37bc2-663f-11e7-834d-ec505990d339.png)

**Comment by teosarca** [comment]
update: API available, see http://w101.metasfresh.com:8081/swagger-ui.html#!/window-rest-controller/getDocumentActionsUsingGET_2

**Comment by wiadev** [comment]
@metasnw 
This is done .

However observed two not quite logical things:

1 . Actions for document-level request and for selected row contains subset of commands duplicated.

Document level - window/540112/1000001/actions
`
{"actions":[{"processId":"ADP_540668","caption":"Neuen Liefervertrag erstellen","description":"","quickAction":false,"defaultQuickAction":false}]}
`

Row level - window/540112/1000001/1/1000000/actions
`
{"actions":[{"processId":"ADP_540668","caption":"Neuen Liefervertrag erstellen","description":"","quickAction":false,"defaultQuickAction":false},{"processId":"ADP_540673","caption":"Vereinbarung Anbauplanung","description":"","quickAction":false,"defaultQuickAction":false}]}
`
As you can see process ADP_540668 is duplicated in both. Looks not quite logical.

How it looks:
![2342341265849](https://user-images.githubusercontent.com/18234060/28881752-1c656cc2-77e4-11e7-95fd-921d74c3d0f8.png)

As you can see (and illustrated above in requirements) we need to have separator between document level actions and row-level actions.

Right now since backend response duplicates document-level actions in row-level actions response we see these items doubled.

Perhaps we can remove this duplication on server side and it will be working correctly.

I've checked with border cases - when there are no row level actions we don't show separator and so forth.

2 . When we invoke process we don't pass rowId as parameter for this process. I'm not sure whether we need this. So the actions are being invoked as other document-level actions without passing the rowId (which contains selected row IDs) as input parameters for process. Please clarify if you need selected row IDs passed as input parameters for process.

3 . Question. Now multiply selections are not supported. E.g. when you select more than 1 row in table - you will not see row-level request sent and actions shown. Please confirm if this is expected behaviour.

**Comment by metasnw** [comment]
> 1 . Actions for document-level request and for selected row contains subset of commands duplicated.

This is only in this particular case like that. The actions are usually different.

> 2 . When we invoke process we don't pass rowId as parameter for this process. 

We do have already processed that work with selection and also with multiple selection. Example: 
https://w101.metasfresh.com:8443/window/540092

only those lines are processed that are selected when you pick the header action "Generate shipment".

**Comment by metasnw** [comment]
IT results:

1. open window b2c
1. upload file
1. open tab documents
1. click on record for uploaded file
1. action menu => download
1. deactivate popup blocker
1. OK: file is downloaded
1. repeat this step: action menu => download.
1. NOK: nothing happens anymore

**Comment by metas-ts** [comment]
cherry-picked to the `release` branch, see https://jenkins.metasfresh.com/job/metasfresh-dist/job/master/3987/console

**Comment by metas-ts** [comment]
cherry picked some more into the `release` branch, see
https://jenkins.metasfresh.com/job/ops/job/cherry-pick-issue-script/212/

**Comment by metasnw** [comment]
IT was ok
